QualityControl/lib/controllers/ObjectController.js

```diff
@@ -79,57 +65,31 @@ export class ObjectController {
-    const { path, validFrom, id, filters } = req.query;
-    if (!path) {
-      updateAndSendExpressResponseFromNativeError(
-        res,
-        new InvalidInputError('Invalid URL parameters: missing object path'),
-      );
-    } else {
-      try {
-        const object = await this._objService.retrieveQcObject(path, Number(validFrom), id, filters);
-        res.status(200).json(object);
-      } catch (error) {
-        updateAndSendExpressResponseFromNativeError(
-          res,
-          new Error('Unable to identify object or read it'),
-        );
-        this._logger.errorMessage(error);
-      }
+    try {
+      const { path, validFrom, filters, id } = req.query;
+
+      const object = await this._objService.retrieveQcObject(path, validFrom, id, filters);
+      res.status(200).json(object);
+    } catch (error) {
+      const responseError = new Error('Failed to retrieve object content');
+
+      logger.errorMessage(`Error validating query parameters: ${error}`);
+      updateAndSendExpressResponseFromNativeError(res, responseError);
     }
   }
 
-  /**
-   * Using `browse` option, request a list of `last-modified` and `valid-from` for a specified path for an object
-   * Use the first `validFrom` option to make a head request to CCDB; Request which will in turn return object
-   * information and download it locally on CCDB if it is not already done so;
-   * From the information retrieved above, use the location with JSROOT to get a JSON object
-   * Use JSROOT to decompress a ROOT object content and convert it to JSON to be sent back to the client for
-   * interpretation with JSROOT.draw
-   * @param {Request} req - HTTP request object with "query" information
-   * @param {Response} res - HTTP response object to provide information on request
-   * @returns {void}
-   */
   async getObjectById(req, res) {
-    const qcgId = req.params?.id;
-    const { validFrom, filters, id } = req.query;
-    if (!qcgId) {
-      updateAndSendExpressResponseFromNativeError(
-        res,
-        new InvalidInputError('Invalid URL parameters: missing object ID'),
-      );
-      return;
-    } else {
-      try {
-        const object = await this._objService.retrieveQcObjectByQcgId(qcgId, id, validFrom, filters);
-        res.status(200).json(object);
-      } catch (error) {
-        updateAndSendExpressResponseFromNativeError(
-          res,
-          new NotFoundError('Unable to identify object or read it by qcg id'),
-        );
-        this._logger.errorMessage(error);
-      }
+    try {
+      const qcgId = req.params?.id;
+      const { validFrom, filters, id } = req.query;
+
+      const object = await this._objService.retrieveQcObjectByQcgId(qcgId, id, validFrom, filters);
+      res.status(200).json(object);
+    } catch (error) {
+      const responseError = new Error('Unable to identify object or read it by qcg id');
+
+      logger.errorMessage(`Error validating query parameters: ${error}`);
```

Comment:
The same here, input is validated in the middleware, thus this is a different kind of error